mathesar_ui/src/stores/table-data/records.ts

```diff
@@ -113,7 +128,7 @@ function preprocessRecords(
           isGroup = true;
         } else {
           // eslint-disable-next-line no-restricted-syntax
-          for (const column of groupColumns) {
+          for (const column of groupingColumnNames) {
```

Comment:
We may want to think about what would be the best way to separate rows. Currently our structure is kept flat and each record is checked with the previous record to separate into groups.

We could evaluate if using `result_indices` would make this cleaner, doing it from a `group -> record` order rather than `record -> group`. This is something we can evaluate as part of https://github.com/centerofci/mathesar/issues/906

Reply:
I had the same thought. I figured once we work on the collapsible groups feature we will probably want to take all the rows within one group and put them into a single element. So the approach your suggesting would make that easier, and I think we could take it up at that point. I wanted to try to keep the changes in this PR somewhat minimal and low risk since I know this PR is already blocking others.

Reply:
Agreed. We can take this up as part of the collapsible groups feature.